src/middle/Stan_math_signatures.ml

```diff
@@ -90,6 +90,14 @@ type fkind =
 type fun_arg = UnsizedType.autodifftype * UnsizedType.t
 type signature = UnsizedType.returntype * fun_arg list * Mem_pattern.t
 
+type variadic_signature =
+  { return_type: UnsizedType.t
+  ; control_args: fun_arg list
+  ; allow_fn_lpdf: bool [@default false]
```

Comment:
I don't think this `allow_lpdf_fn` field serves any purpose. Allowing lpdfs really only works for reduce_sum which was special-cased anyway.
That's because the user can wrap a normal lpdf in a plain function so `allow_lpdf` is really more like `allow_lupdf` and _that_ is very hard to get right. Even something like `reduce_log_sum_exp`, when applied to `binomial_lupdf` would drop the binomial coefficients and often return nonsense.

Reply:
You’re right, I was originally hopeful that I could fit reduce_sum into this paradigm but this flag ended up not being useful.